src/x3d-parser.js

```diff
@@ -545,19 +545,24 @@ function renderX3D(THREE, x3dXml, scene, useImageTexture) {
 
         var tree = { 'string': 'Scene', children: [] };
 
-        parseChildren(x3dXml.documentElement.childNodes[1], tree);
-
-        return tree;
+        for(var i = 0; i < x3dXml.documentElement.childNodes.length; i++) {
+            if(x3dXml.documentElement.childNodes[i].nodeName === 'Scene') {
+                parseChildren(x3dXml.documentElement.childNodes[i], tree);
+                return tree;
+            }
+        }
+        console.error("Unable to find Scene element in X3D document")
 
     };
 
     var parseChildren = function (parentNode, parentResult) {
         for (var i = 0; i < parentNode.childNodes.length; i++) {
             var currentNode = parentNode.childNodes[i];
             if (currentNode.nodeType !== 3) {
+                var nodeAttr = currentNode.attributes[0] || {}
                 var newChild = {
                     'nodeType': currentNode.nodeName.toLocaleLowerCase(),
-                    'string': getNodeGroup(currentNode.nodeName) + ' ' + currentNode.nodeName.toLocaleLowerCase(),
```

Comment:
I'm not sure if this is a bug in the initial implementation, or if there are just varying version of the X3D spec, but this code was failing to find the contents / tags of `USE` and `DEF` attributes.

In my X3D models, I was able to fix this by directly adding the attributes like this.